src/dag-builder/file/index.js

```diff
@@ -64,6 +66,83 @@ async function * buildFile (source, ipld, options) {
   }
 }
 
+const serialize = (node, ipld, options) => {
+  if (!options.codec && node.length) {
+    options.cidVersion = 1
+    options.codec = 'raw'
+  }
+
+  if (isNaN(options.hashAlg)) {
+    options.hashAlg = mh.names[options.hashAlg]
+  }
+
+  if (options.hashAlg !== mh.names['sha2-256']) {
+    options.cidVersion = 1
+  }
+
+  if (options.format) {
+    options.codec = options.format
+  }
+
+  const format = mc[options.codec.toUpperCase().replace(/-/g, '_')]
+
+  return ipld.serialize(node, format, options)
+}
+
+async function * buildFileBatch (source, ipld, options) {
+  let count = -1
+  let previous
+  let nodesToPersist = []
+
+  const timer = setIntervalAsync(async () => {
+    const temp = nodesToPersist
+    nodesToPersist = []
+    await ipld.putBatch(temp, options)
```

Comment:
Where does the error go if this throws?  Should we retry the batch?  What if retrying fails?